lib/bitcoin/BitcoinClient.ts

```diff
@@ -313,6 +325,156 @@ export default class BitcoinClient {
     return transaction;
   }
 
+  /**
+   * Calculates an estimated fee for the given transaction. All the inputs and outputs MUST
+   * be already set to get the estimate more accurate.
+   *
+   * @param transaction The transaction for which the fee is to be calculated.
+   */
+  private async calculateTransactionFee (transaction: Transaction): Promise<number> {
+    // Get esimtated fee from RPC and add some percentage to it.
+    const estimatedFeeInKb = await this.getCurrentEstimatedFeeInSatoshisPerKb();
+    const estimatedFeeInKbWithPercentage = estimatedFeeInKb + (estimatedFeeInKb * .2);
+
+    // Estimate the size of the transaction and add some percentage to it.
+    const estimatedSizeInBytes = (transaction.inputs.length * 150) + (transaction.outputs.length * 50);
+    const estimatedSizeInBytesWithPercentage = estimatedSizeInBytes + (estimatedSizeInBytes * .2);
+    const estimatedSizeInKb = estimatedSizeInBytesWithPercentage / 1000;

```

Comment:
Consider just apply one percentage at the very end, as opposed to applying percentage twice in line 337 and 341.

Reply:
The math is not the same if we just multiply w/ .4 at the very end. I have moved the percentage additions to the end.